features/eolearn/features/doubly_logistic_approximation.py

```diff
@@ -0,0 +1,82 @@
+"""
+import itertools as it
+import numpy as np
+from scipy.optimize import curve_fit
+from eolearn.core import EOTask
+
+
+def doubly_logistic(middle, initial_value, scale, a1, a2, a3, a4, a5):
+    # pylint: disable=invalid-name,locally-disabled
+    """
+    Function that is passed to scipy.optimize
+    """
+    return initial_value + scale * np.piecewise(middle, [middle < a1, middle >= a1],
+                                                [lambda y: np.exp(-((a1 - y) / a4) ** a5),
+                                                 lambda y: np.exp(-((y - a1) / a2) ** a3)])
+
+
+class DoublyLogisticApproximationTask(EOTask):
+    """
+    EOTask class for calculation of doubly logistic approximation on each pixel for a feature. The task creates new
+    feature with the function parameters for each pixel as vectors.
+    :param feature: A feature on which the function will be approximated
+    :type feature: str
+    :param new_feature: Name of the new feature where parameters of the function are saved
+    :type new_feature: str
+    :param initial_parameters: Initial parameter guess
+    :type initial_parameters: List of floats length 7 corresponding to each parameter
+    """
+
+    def __init__(self, feature, new_feature='DOUBLY_LOGISTIC_PARAM', initial_parameters=None, mask_data=False):
+        self.feature = feature
+        self.initial_parameters = initial_parameters
+        self.new_feature = new_feature
+        self.mask_data = mask_data
+
+    def _fit_optimize(self, x_axis, y_axis):
+        """
+        :param x_axis: Horizontal coordinates of points
+        :type x_axis: List of floats
+        :param y_axis: Vertical coordinates of points
+        :type y_axis: List of floats
+        :return: List of optimized parameters [c1, c2, a1, a2, a3, a4, a5]
+        """
+        bounds_lower = [np.min(y_axis), -np.inf, x_axis[0], 0.15, 1, 0.15, 1, ]
+        bounds_upper = [np.max(y_axis), np.inf, x_axis[-1], np.inf, np.inf, np.inf, np.inf, ]
+        if self.initial_parameters is None:
+            self.initial_parameters = [np.mean(y_axis), 0.2, (x_axis[-1] - x_axis[0]) / 2, 0.15, 10, 0.15, 10]
+        optimal_values = curve_fit(doubly_logistic, x_axis, y_axis, self.initial_parameters,
+                                   bounds=(bounds_lower, bounds_upper), maxfev=1000000,
+                                   absolute_sigma=True)
+        return optimal_values[0]
+
+    def execute(self, eopatch):
+        """
+        :param eopatch: Input eopatch with data on which the doubly logistic approximation is computed
+        :return: Output patch with doubly logistic approximation parameters
+        """
+        data = eopatch.data[self.feature].squeeze(axis=3)
+
+        times = np.asarray([time.toordinal() for time in eopatch.timestamp])
+        times = (times - times[0]) / (times[-1] - times[0])
+
+        time, height, width = data.shape
+
+        if self.mask_data:
+            valid_data_mask = eopatch.mask['VALID_DATA']
```

Comment:
Please avoid hardcoding feature names. I suggest to change `mask_data` (bool) argument to `mask_feature` (FeatureType, feature_name) to make it more generalizable.